src/server/mod.rs

```diff
@@ -308,6 +308,10 @@ impl ServerConfig {
     /// Sets a single certificate chain and matching private key.  This
     /// certificate and key is used for all subsequent connections,
     /// irrespective of things like SNI hostname.
+    /// ***Note that the end-entity certificate must have the 
+    /// [Subject Alternative Name](https://en.wikipedia.org/wiki/Subject_Alternative_Name) 
```

Comment:
Let's avoid using Wikipedia as a reference. RFC 6125 is a much better reference.

Also, I don't think this needs to be called out in bold or whatever `***` indicates.

Reply:
Since multiple people were confused about this requirement within a short time frame (see PR OP), I thought the emphasis markup was warranted. Removed it now.